geode-core/src/main/java/org/apache/geode/internal/cache/PoolManagerImpl.java

```diff
@@ -63,6 +64,11 @@ public static PoolManagerImpl getPMI() {
     return result;
   }
 
+  @TestOnly
+  public static void setImpl(PoolManagerImpl poolManager) {
+    impl = poolManager;
+  }
+
```

Comment:
I think we're going in the wrong direction with some of these TestOnly setters, especially when the constant or variable is annotated with `@MakeNotStatic` and it needs to have `final` removed to make it work.

Reply:
I agree, I would love to work with you to make these tests better.